src/eventcontroller/EventController.ts

```diff
@@ -1,16 +1,34 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import EventObserver from '../eventobserver/EventObserver';
+import EventReporter from '../eventreporter/EventReporter';
 import AudioVideoEventAttributes from './AudioVideoEventAttributes';
 import DeviceEventAttributes from './DeviceEventAttributes';
 import EventName from './EventName';
 import MeetingHistoryState from './MeetingHistoryState';
 
 /**
- * [[EventController]] keeps track of events and notifies audio-video observers.
- * An event describes the success and failure conditions for the meeting session.
+ * [[EventController]] keeps track of a list of event observers and notifies them of SDK events.
+ * An event can describe success or failure of SDK components that make use of this controller.
+ * Example: The success and failure for a meeting session.
```

Comment:
As we have chosen (I think it makes sense to, but not super familiar), to make this injectable, can we call out why or why not a builder may want to inject their own or more likely just use the default implementation?

Reply:
Hmm, I think that might be better for a blog post or something similar. We don't typically document why a builder might want to inject their own version of an interface.

Reply:
I think many builders would like to live off doc strings rather then possibly out of date blog posts. I'm also just thinking a few sentences. If we don't want builders injecting we should call that out too. @ltrung @simmkyu what do you think?

> We don't typically document why a builder might want to inject their own version of an interface.

'We don't currently do it' !== 'Good practice' lol.

Reply:
I don't think I disagree, but I feel it would feel out of place if we were to document it on some interfaces but not others. I'll wait to hear what @ltrung  and @simmkyu think.

Reply:
I think we try to document more on the interface if possible now. At least @richnew10 used to get on me all the time for that :)